client/components/date-picker/docs/example.jsx

```diff
@@ -1,60 +1,111 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	PureRenderMixin = require( 'react-pure-render/mixin' );
+import React, { Component } from 'react';
 
 /**
  * Internal dependencies
  */
-var Card = require( 'components/card' ),
-	DatePicker = require( 'components/date-picker' );
+import Card from 'components/card';
+import DatePicker from 'components/date-picker';
 
 /**
  * Date Picker Demo
  */
-var datePicker = React.createClass( {
-	mixins: [ PureRenderMixin ],
-	displayName: 'DatePicker',
-
-	getInitialState: function() {
-		var date = new Date();
-		date.setDate( date.getDate() + 3 );
-		date.setMilliseconds( 0 );
-		date.setSeconds( 0 );
-		date.setMinutes( 0 );
-		date.setHours( 0 );
-
-		return {
-			events: [
-				{
-					title: '1 other post scheduled',
-					date: new Date( '2015-07-15 10:30' ),
-					type: 'scheduled'
-				},
-				{
-					title: 'Happy birthday Damian',
-					date: new Date( '2015-07-18 15:00' ),
-					type: 'birthday'
-				},
-				{
-					title: 'Do not rest',
-					date: new Date( '2015-07-18 8:00' )
-				}
-			],
-			selectedDay: this.moment( date )
-		};
-	},
-
-	selectDay: function( date, modifiers ) {
+class DatePickerExample extends Component {
+	state = {
+		events: [
+			{
+				title: 'Today',
+				date: new Date(),
+				type: 'scheduled'
+			},
+
+			{
+				title: 'Social Media - Facebook',
+				date: new Date(),
+				socialIcon: 'facebook',
+			},
+
+			{
+				title: 'Social Media - Twitter',
+				date: new Date(),
+				socialIcon: 'twitter',
+				socialIconColor: false,
+			},
+
+			{
+				title: 'Social Media - Google Plus',
+				date: new Date(),
+				socialIcon: 'google-plus',
+			},
+
+			{
+				title: 'Social Media - LinkedIn',
+				date: new Date(),
+				socialIcon: 'linkedin',
+			},
+
+			{
+				title: 'Social Media - Tumblr',
+				date: new Date(),
+				socialIcon: 'tumblr',
+			},
+
+			{
+				title: 'Social Media - Path',
+				date: new Date(),
+				socialIcon: 'path',
+				socialIconColor: false,
+			},
+
+			{
+				title: 'Social Media - Eventbrite',
+				date: new Date(),
+				socialIcon: 'eventbrite',
+				socialIconColor: false,
+			},
+
+			{
+				title: 'Gridicon - Time',
+				date: new Date(),
+				icon: 'time',
+			},
+
+			{
+				title: 'Gridicon - Offline',
+				date: new Date(),
+				icon: 'offline',
+			},
+
+			{
+				title: 'Gridicon - Shipping',
+				date: new Date(),
+				icon: 'shipping',
+			},
+
+			{
+				title: 'Tomorrow is tomorrow',
+				date: new Date( +new Date() + 60 * 60 * 24 * 1000 ),
+				type: 'future-event',
+			},
+			{
+				title: 'Yesterday',
+				date: new Date( +new Date() - 60 * 60 * 24 * 1000 ),
+				type: 'past-event',
+			}
+		]
+	};
+
+	selectDay( date, modifiers ) {
```

Comment:
This throws an error when clicking because it is not bound to the definition site. If you change this line to `selectDay = ( date, modifiers ) => {` then it should work.